modelling/src/main/java/org/axonframework/modelling/saga/AbstractSagaManager.java

```diff
@@ -243,8 +243,8 @@ public boolean canResolve(ScopeDescriptor scopeDescription) {
 
         private static <T> T newInstance(Class<T> type) {
             try {
-                return type.newInstance();
-            } catch (InstantiationException | IllegalAccessException e) {
+                return type.getDeclaredConstructor().newInstance();
```

Comment:
I believe the `Class#newInstance` call has been deprecated in Java 9 (as can be found [here](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Class.html#newInstance())).
The framework is however based on Java 8 at the moment, so I do not feel this change should be made at this point.

Additionally, if I perform this change on an Axon project using Java 8, this line breaks as the `getDeclaredConstructor()` expects parameter types to be inserted.

Reply:
Disregard this, I am having a bad case of the Mondays (aka, unable to correctly read exceptions from my IDE).